### DISCUSSION TITLE
Remchem feeds not picked up by njsPC

### DISCUSSION THREAD

**Discussion by lockeandev** [tagyoureit/nodejs-poolController#645] [Q&A] [answered]
I had ph, orp, and temp feeds from atlas industrial probes, and a flow switch in a wet well, configured and sending values to ny njsPC instance. I was working in the pool shed the other day and jostled the Raspi and it fell out of it's mount. It didn't fall to the ground but was kept suspended by the 5-6 cable connections to it. I double checked all the connections, but ever since then, dash panel is showing zeros for all the sensor values and no flow is detected even when the flow switch is physically engaged.  I can view values for temp, ph, and orp just fine in the remchem interface, only the gpio pin status of the flow switch is not reported.

So there's two issues:

Any idea on why my njsPC instance is not getting the ph, orp, and temp values anymore? 

And how can I troubleshoot the gpio issue? I have checked the connections and they look ok to me, but the pin status in remchem does not update with changes in flow.

**Comment by rstrouse** [comment]
Double check the configuration in dashPanel for the hardware to make sure it is communicating with REM.  You can verify this by hitting the dropdown on the devices list for the flow switch.  If it is then check to see if the pin is orange when tripped.  If it isn't restart REM and watch the logs to make sure the pin is still responding.

![image](https://user-images.githubusercontent.com/47839015/206929668-558a1c65-a33f-4bee-ab86-e128554c4b77.png)

**Reply by lockeandev** [reply]
That's the thing. I think the communication between dashPanel and REM is broken. The pin is not orange. I do have values that update on dashPanel for the i2c bus devices. 

I've restarted the whole pi, and no change in behavior. What logs should I looks at and what should I look for?

**Reply by rstrouse** [reply]
If you are running pm2 run `pm2 monit` and it will show all the logs.

**Reply by lockeandev** [reply]
k, here are my logs, I'm not sure why MQTT is disconnected, the credentials for my MQTT server are correct, and I see unhandledpromiseexception over and over.

[REM-error.log](https://github.com/tagyoureit/nodejs-poolController/files/10203629/REM-error.log)
[REM-out.log](https://github.com/tagyoureit/nodejs-poolController/files/10203630/REM-out.log)

**Reply by rstrouse** [reply]
Double check pin 11 in REM to make sure it is active.  The exception is because the pin isn't initialized.

**Reply by lockeandev** [reply]
it is active. I restarted REM via PM2 and the exception is gone, now. 
<img width="875" alt="image" src="https://user-images.githubusercontent.com/11232490/206939234-d88c8413-2950-4f11-b3fe-58ae34867f5b.png">

dashpanel for njsPC still shows not slow and the pin status is not updated. I will go and check the physical connection again. 

the logs now just have a bunch of GETs  not sure why all of those gets are there. everyone of them is an ip for a samsung device including tvs that should be connecting to dashpanel in anyway. 

<img width="456" alt="image" src="https://user-images.githubusercontent.com/11232490/206939372-d4564fd3-9c7e-4f1e-a507-9fa91bc1407d.png">

**Reply by rstrouse** [reply]
That is your Samsung device asking what is at the endpoint.  This is part of the UPnP queries from the TV.  These are normal for this type of device.

What does that feed do for your flow switch?  Are you turning on a circuit with the flow switch?

**Comment by lockeandev** [comment]
I'm just using the flow switch to have acid dosed when there is flow.

**Reply by rstrouse** [reply]
But you don't need to feed a circuit to do that.

**Reply by lockeandev** [reply]
Then I guess I don't understand how feeds and triggers are used. But I didn't make any changes. All the values from rem chem just stopped being sent to njPC. 

Do I need a feed for ph or orp to be sent to njPC?

![Screenshot_20221212_194348_Chrome](https://user-images.githubusercontent.com/11232490/207213728-72b3db5f-261c-4bfb-9ba6-60187198701a.jpg)

**Reply by rstrouse** [reply]
Yes the key difference between these two is that the flow switch is polled when it cycles.  It will make sure all the conditions are true while dosing and before starting a dose.  This makes sure there are no hardware or connection failures between the servers and the equipment.

This is all contained within the REM Chem controller code to verify the switch.  It also does this for some of the other options such as Flow Only readings.  pH and ORP readings when there is nothing going through the flow cell are pretty useless since it will only be the levels for that cup of water in the cell.  Oh and that reminds me I need to add acid to my tank and throw a gallon of chlorine in before the next storm.
![image](https://user-images.githubusercontent.com/47839015/207216766-ca04e361-7924-4289-927f-e1adebcb8351.png)

**Comment by lockeandev** [comment, marked as answer]
So is there anything else I can look at to see why ph and orp info is shown
in Remchem but not in njsPC?

All of my mqtt topics for ph and orp are also showing zero. Pump vsf is
being feed from njsPC to mqtt topics just fine.

Salt is also showing as 0 ppm on dashPanel:

[image: image.png]

**Reply by rstrouse** [reply]
Change your feed from pH to all values.  It doesn't matter for ORP but the pH probe has an associated temperature reading.  If you have Flow readings only selected it should only be accepting the ORP/pH feed when the flow switch is showing flow.  This is so the calculations are accurate.  You will find the pH in a flow cell without flow can be dramatically different than it is when the cell is flowing.

Btw your image is not showing.
